ibis/pandas/execution/generic.py

```diff
@@ -428,8 +428,12 @@ def execute_aggregation_dataframe(op, data, scope=None, **kwargs):
         for metric in op.metrics
     ]
 
-    # group by always needs a reset to get the grouping key back as a column
-    result = pd.concat(pieces, axis=1).reset_index()
+    result = pd.concat(pieces, axis=1)
+
+    # If grouping, need a reset to get the grouping key back as a column
+    if op.by:
```

Comment:
Can you explain why this change have no user facing effect? Before this PR, is the column "index" being dropped sometime later?

Reply:
Yep, the Ibis Pandas client uses [`ibis.pandas.core.execute_and_reset`](https://github.com/ibis-project/ibis/blob/13387c5a0969c99be403af5b3e8aa70bdb671623/ibis/pandas/core.py#L418), which gets rid of columns that are not in the schema.

(I don't think that function is directly intended to correct the output of node execution methods that return unnecessary columns. It's main goal is to reset the index of the final dataframe before returning it to the user, keeping it as a column if its in the schema, and dropping it otherwise. This just happens to fix the output of `execute_aggregation_dataframe`.)

Reply:
Yeah I think it's cleaner if the output of `execute_aggregation_dataframe` doesn't have a column called "index"